### PR TITLE
Add pkg linter

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
## Scope of the PR

* Added https://github.com/QuiiBz/sherif
* Updated vite & vitests

## Checklist

- [x] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/search/package.json [hunk 1]
```diff
@@ -45,20 +45,19 @@
     "@trpc/next": "10.43.1",
     "@trpc/react-query": "10.43.1",
     "@trpc/server": "10.43.1",
-    "@types/debug": "^4.1.7",
-    "@urql/exchange-auth": "^2.1.4",
+    "@urql/exchange-auth": "2.1.4",
     "algoliasearch": "4.23.3",
     "clsx": "^1.2.1",
```
Comment (lkostrowski) [inline apps/search/package.json:50]: Shouldnt you pin all dependencies? I see only some of them are pinned

**Review by witoszekdev — APPROVED** [review]

**Review by lkostrowski — CHANGES_REQUESTED** [review]
can you pin all packages? Now its partially pinned

**Review by lkostrowski — CHANGES_REQUESTED** [review]
Maybe we can also add this rule
https://github.com/kellyselden/eslint-plugin-json-files/blob/main/docs/rules/restrict-ranges.md

or the new linter will handle this? I think from now on, PR should fail to build if packages are not pinned

Also, we should probably fix typescript first and rebase

**Comment by krzysztofzuraw** [comment]
We decided to move pinning versions to other PR

**Comment by krzysztofzuraw** [comment]
I'm closing this PR - I will first install linter and in other PR I will this its errors

**Pull request closed without merging**